### DISCUSSION TITLE
How to get the normal vector to a sideset( nodeset)?

### DISCUSSION THREAD

**Discussion by Kavan-Khaledi** [idaholab/moose#22377] [Q&A Modules: General] [answered]
Dear MOOSE team,

I am using the "CoupledVarThresholdElementSubdomainModifier" to simulate the boundary movement between two blocks. The velocity of the moving boundary (let's say V) is controlled by a coupled variable c (in my example: mass concentration). However, the direction of the movement should be perpendicular to the boundary. Therefore, the boundary-moving function should look something like this:

 **X**=**X0**+V(c)* t * [nx, ny, nz], 

where [nx, ny, nz] is the normal vector to the boundary at point (x,y,z) and time t. 

Is there a userobject or auxiliary variable in Moose that I can use to find the normal vector? If not, what is your advice?

Best regards,
Kavan

**Comment by GiudGiud** [comment, marked as answer]
hello

You can get the sides from the _current_elem (a class attribute to this user object), with a loop on the sides using this routine to retrieve them.
https://libmesh.github.io/doxygen/classlibMesh_1_1Elem.html#a9c669d37f9a794d933bae07babc2eb60

then from these you can get the normal (with a cross product for example), once you've checked that the side is part of the boundary you care about with something like this for example:
```
https://libmesh.github.io/doxygen/classlibMesh_1_1BoundaryInfo.html#a99bb67d6ae6c470d2cbfdb232dae609e
(this one in particular uses the element and the side index, not the side element)
```

Guillaume
